fulcio: AWS KMS support

sfox-equinix opened the pull request:
## Description of the change

This commit adds support for AWS KMS keys to the fulcio chart. The following variables were added:

- `kmsType` - The KMS type (e.g., `aws`)
- `awsKmsCredentialsSecretName` - The name of an existing Kubernetes secret containing the IAM credentials to authenticate with AWS KMS
  - The secret should contain two key value pairs: `accessKeyId` and `secretAccessKey`
- `awsKmsRegion` - The AWS region string where the KMS key is hosted

Users must set `kmsType` to the value `aws` in order to use this feature.

## Existing or Associated Issue(s)

N/A

## Additional Information

I tested these changes using a privately-hosted Sigstore instance. To use AWS KMS keys, users will need to supply the following changes via the values files:

- Set `kmsType` to `aws` in values.yaml
- Set `certificateAuthority` to a `kmsca` in values.yaml
- Set `kms_resource` to a `awskms://` string in values.yaml
- Set `kms_cert_chain` to a PEM blob containing the signers certificate (should be the first certificate in the file), optionally followed by any parent certificates
- Set `awsKmsRegion` to the desired region string in values.yaml
- Create a Kubernetes secret containing the AWS IAM credentials. It should be named using the same string passed to the `awsKmsCredentialsSecretName`

## Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). Where applicable, update and bump the versions in any associated umbrella chart
- [x] Variables are documented in the `values.yaml` and added to the README.md. The [helm-docs](https://github.com/norwoodj/helm-docs) utility can be used to generate the necessary content. Use `helm-docs --dry-run` to preview the content.
- [x] JSON Schema generated.
- [ ] List tests pass for Chart using the [Chart Testing](https://github.com/helm/chart-testing) tool and the `ct lint` command. (**Note: This fails for me with the error: `Error loading configuration: 'chart_schema.yaml' neither specified nor found in default locations`)

vipulagarwal reviewed (approved):
LGTM

bobcallaway commented:
can we please split the changes out into two PRs (one for fulcio, one for rekor)? otherwise the release logic won't be happy.

sfox-equinix commented:
Hi @bobcallaway - I split the rekor changes out into this pull request: https://github.com/sigstore/helm-charts/pull/832

I rebased the current pull request to be for fulcio only.

bobcallaway commented on charts/fulcio/README.md:
```diff
@@ -121,6 +121,8 @@ helm uninstall [RELEASE_NAME]
 | server.args.grpcPort | int | `5554` |  |
 | server.args.hsm_caroot_id | string | `nil` |  |
 | server.args.port | int | `5555` |  |
+| server.awsKmsCredentialsSecretName | string | `"aws-kms-credentials"` | ubernetes secret name containing IAM credentials for use with AWS KMS |
```
```suggestion
| server.awsKmsCredentialsSecretName | string | `"aws-kms-credentials"` | kubernetes secret name containing IAM credentials for use with AWS KMS |
```

bobcallaway reviewed (approved):
one spelling nit but LGTM
